mantidimaging/core/operations/median_filter/median_filter.py

```diff
@@ -83,19 +82,34 @@ def filter_func(data: ImageStack, size=None, mode="reflect", progress=None, forc
         :return: Returns the processed data
 
         """
+        # Validation
         h.check_data_stack(data)
-
-        if not size or not size > 1:
+        if size is None or size <= 1:
             raise ValueError(f'Size parameter must be greater than 1, but value provided was {size}')
 
+        params = {'mode': mode, 'size': size, 'force_cpu': force_cpu, 'progress': progress}
         if not force_cpu:
-            _execute_gpu(data.data, size, mode, progress)
+            ps.run_compute_func(MedianFilter.cuda_compute_function, data.data.shape[0], data.shared_array, params)
         else:
-            _execute(data, size, mode, progress)
+            ps.run_compute_func(MedianFilter.compute_function, data.data.shape[0], data.shared_array, params)
 
-        h.check_data_stack(data)
         return data
 
+    @staticmethod
+    def compute_function(i: int, array: np.ndarray, params: Dict[str, Any]):
+        mode = params['mode']
+        size = params['size']
+
+        array[i] = _median_filter(array[i], size=size if size is not None else 3, mode=mode)
```

Comment:
`size = size if size is not None else 3` -> `size = size` (Because None is caught outside compute fuction)